### ISSUE TITLE
tutorial id not coming through on webhook request

### ISSUE THREAD

**Issue by moT01** [freeCodeCamp/CodeAlly-CodeRoad-freeCodeCamp#37]
I did some tests to see if the webhooks were working. I couldn't get the tutorial id to come through with the request. My setup was to use ngrok as the webhook URL which forwarded to my fCC api locally.

Code in my api:
<img width="406" alt="Screen Shot 2022-01-08 at 9 39 53 AM" src="https://user-images.githubusercontent.com/20648924/148650400-41904701-0e13-43b5-b4bd-55ad957746e9.png">
 
 First attempt - on CodeAlly + CodeRoad 0.18.4:
<img width="864" alt="Screen Shot 2022-01-07 at 2 54 43 PM" src="https://user-images.githubusercontent.com/20648924/148650417-465c66ef-c143-4e05-98a0-0bcecff671c8.png">
^ tutorial id is empty, user token came through

Second attempt - Local Docker container with CodeRoad 0.15.1:
<img width="726" alt="Screen Shot 2022-01-07 at 7 59 37 PM" src="https://user-images.githubusercontent.com/20648924/148650484-97cd2d87-894e-4cf3-837a-6852442db2d5.png">
^ the tutorial id came through - I didn't add the token to the env so that wasn't there. Made another try with 0.17.4 same result.

Other attempts:
Attempted to downgrade CodeRoad to 0.15.1 in CodeAlly - couldn't get the tutorial to load
Switched back to 0.18.4 - view the logs, didn't see anything particularly helpful in them. This was in there:
```
Using tutorial url from env: https://raw.githubusercontent.com/freeCodeCamp/learn-celestial-bodies-database/main/tutorial.json
Tutorial: Celestial Bodies Database (1.0.2)
```

Final attempt - was able to downgrade to 0.17.4 on CodeAlly and run the tutorial. Same result as my other attempts on CodeAlly:
<img width="458" alt="Screen Shot 2022-01-08 at 10 07 53 AM" src="https://user-images.githubusercontent.com/20648924/148651274-4a17977d-2a8e-4e9a-8d69-309e34f977fc.png">
^ tutorial id is blank still. So 0.17.4 worked locally, but not on CodeAlly. Perhaps it's something to do with me using ngrok? I dunno, the request still came through, it just didn't have that id. Any ideas?

**Comment by ShMcK** [comment]
I've put a PR to log the webhook success/failure attempts along with the body.
It should help debug the issue.

https://github.com/coderoad/coderoad-vscode/pull/562

**Comment by ShMcK** [comment]
@moT01 I think I found the issue, though I don't know the cause.

There is no `id` on the tutorial on CodeAlly.

You can see it for yourself by running `cat ../../codeally/.local/coderoad_tutorial.json`, which shows the tutorial data. There is no id.

**Comment by AdamZaczek** [comment]
Does this mean this is something to look at on our side @ShMcK ?
I'm not aware when do we get those ids if so.

**Comment by ShMcK** [comment]
@moT01 just add an id like "fcc/learn-celestial-bodies-database' as an id manually, that should work for now. Then run build.

For example, here: https://github.com/freeCodeCamp/learn-celestial-bodies-database/blob/main/coderoad.yaml

I don't have a way to generate ids at this time.

**Comment by moT01** [comment]
Would that be just another field in the main config there? ...
```
  repo:
    uri: https://github.com/freeCodeCamp/learn-celestial-bodies-database
    branch: v1.0.2
  id: fcc/learn-celestial-bodies-database:v1.0.2
```

**Comment by moT01** [comment]
This works @ShMcK. I added the id to the top level of the yaml:
![Screen Shot 2022-01-09 at 4 30 09 PM](https://user-images.githubusercontent.com/20648924/148703747-5f2ce0e3-59a8-47b3-afec-1950cb4dcea7.png)

The CLI tools give me an error:
![Screen Shot 2022-01-09 at 4 31 24 PM](https://user-images.githubusercontent.com/20648924/148703788-448efc8b-b448-4631-88ed-59dcb24713aa.png)

I bypassed that by adding the id directly to the `tutorial.json` file, tested it out on CodeAlly - and the id came through on the back end. So this looks like it works 👍

**Comment by ShMcK** [comment]
Great, I'll look into the issue with the CLI and ids

**Comment by AdamZaczek** [comment]
Let me know once this is fixed. We will update the coderoad to the newest version so this issue and https://github.com/freeCodeCamp/CodeAlly-CodeRoad-freeCodeCamp/issues/39 can be closed.

**Comment by AdamZaczek** [comment]
Let me know once this is fixed. We will update the coderoad in tutorials to the newest version so we can close this issue and #39.

**Comment by ShMcK** [comment]
@moT01 if you can publish the tutorials with ids, that should be sufficient for now.

**Comment by moT01** [comment]
Yes, I plan to add the id's to all the tutorials - and that worked in my testing. I am holding off on it for now because it's not support by the CLI tools yet (as of 4 days ago). So I would have to remove them before running `coderoad build`, then re-add them. I don't mind manually adding them. Was just going to do that last - or close to last.

**Comment by ShMcK** [comment]
Tutorial ids should be included in the cli build with `@coderoad/cli v0.9.1`. Let me know if there's any issue!

**Comment by moT01** [comment]
I added the id's to all the tutorials. Still a little curious why they weren't being sent with the request on CodeAlly, but were locally. Doesn't matter, this should work.
